ghost/core/core/server/data/migrations/versions/5.56/2023-07-15-10-11-12-update-members-email-disabled-field.js

```diff
@@ -0,0 +1,22 @@
+const logging = require('@tryghost/logging');
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Setting email_disabled to true for all members that have their email on the suppression list');
+
+        await knex('members')
+            .join('suppressions', 'members.email', 'suppressions.email')
+            .update({
+                email_disabled: true
+            });
+    },
+    async function down(knex) {
```

Comment:
is there something that would prevent a race condition here where a member on suppressions list gets set to `true` and then the general one runs and it gets set to `false`? 

I don't know the actual migration process but "up" suggests that runs first, and then "down" runs at the end, but that seems to me it would undo what was done in up?

Reply:
Ah nice, thanks for clarifying 👍